Library/ViewModels/DiscoveryProjectCardViewModelTests.swift

```diff
@@ -119,4 +123,120 @@ final class DiscoveryProjectCardViewModelTests: TestCase {
 
     self.projectImageUrlString.assertValues(["http://www.kickstarter.com/full.jpg"])
   }
+
+  func testLoadProjectTags_ProjectIsPWL_CategoryIsRoot() {
+    let project = Project.template
+      |> \.staffPick .~ true
+      |> \.category .~ .illustration
+
+    self.loadProjectTags.assertDidNotEmitValue()
+
+    self.vm.inputs.configure(with: (project, .art, nil))
+
+    self.loadProjectTags.assertValues([
+      [
+        DiscoveryProjectTagPillCellValue(
+          type: .green,
+          tagIconImageName: "icon--small-k",
+          tagLabelText: "Projects We Love"
+        ),
+        DiscoveryProjectTagPillCellValue(
+          type: .grey,
+          tagIconImageName: "icon--compass",
+          tagLabelText: "Illustration"
+        )
+      ]
+    ])
+  }
+
+  func testLoadProjectTags_CategoryIsNil() {
+    let project = Project.template
+      |> \.staffPick .~ true
+      |> \.category .~ .illustration
+
+    self.loadProjectTags.assertDidNotEmitValue()
+
+    self.vm.inputs.configure(with: (project, nil, nil))
+
+    self.loadProjectTags.assertValues([
+      [
+        DiscoveryProjectTagPillCellValue(
+          type: .green,
+          tagIconImageName: "icon--small-k",
+          tagLabelText: "Projects We Love"
+        ),
+        DiscoveryProjectTagPillCellValue(
+          type: .grey,
+          tagIconImageName: "icon--compass",
+          tagLabelText: "Illustration"
+        )
+      ]
+    ])
+  }
+
+  func testLoadProjectTags_CategoryMatchesProjectCategory() {
+    let project = Project.template
+      |> \.staffPick .~ true
+      |> \.category .~ .illustration
+
+    self.loadProjectTags.assertDidNotEmitValue()
+
+    self.vm.inputs.configure(with: (project, .illustration, nil))
+
+    self.loadProjectTags.assertValues([
+      [DiscoveryProjectTagPillCellValue(
+        type: .green,
+        tagIconImageName: "icon--small-k",
+        tagLabelText: "Projects We Love"
+      )]
+    ], "Does not show subcategory tag when filtered category is the same as project subcategory")
+  }
+
+  func testTagsCollectionViewHidden_WhenProjectIsNotStaffPick_CategoryTagShouldShow() {
```

Comment:
Should we be asserting the values of `loadProjectTags` in these other tests?

Reply:
Hm so these tests are specifically for the `tagCollectionViewHidden` output, the tests above cover all the possible states of `loadProjectTags`.